File: src/main/java/practice/email/clustering/TokenEditDistanceFunction.java
package practice.email.clustering;

import practice.email.parser.TokenEditDistanceKt;
import weka.core.Instances;

public class TokenEditDistanceFunction extends AbstractStringDistanceFunction {
    public TokenEditDistanceFunction() {
    }

    public TokenEditDistanceFunction(Instances data) {
        super(data);
    }

    @Override
    public String globalInfo() {
        return null;
    }

    @Override
    public String getRevision() {
        return null;
    }

    double stringDistance(String stringA, String stringB) {
        return (double) TokenEditDistanceKt.getEditDistance(stringA, stringB);
    }
}

File: old_srcs/quoteParser_old/src/main/kotlin/practice/email/clustering/TokenEditDistanceFunction.kt
package practice.email.clustering

import practice.email.parser.*
import weka.core.Instances

class TokenEditDistanceFunction : AbstractStringDistanceFunction {
    constructor() {
    }

    constructor(data: Instances) : super(data) {
    }

    override fun globalInfo(): String? {
        return null
    }

    override fun getRevision(): String? {
        return null
    }

    override fun stringDistance(stringA: String, stringB: String): Double {
        return getEditDistance(stringA, stringB).toDouble()
    }
}
